contracts/Core/AssetManager.sol

```diff
@@ -306,6 +282,124 @@ contract AssetManager is ACL, AssetStorage {
         );
     }
 
+    function addPendingCollections() 
+        external
+        onlyRole(parameters.getAssetConfirmerHash()) 
+    {
+        if(numPendingCollections != 0)
+        {
+            
+            uint256 temp = numPendingCollections;
+            for(uint8 i = 1; i <= temp; i++){
+                uint256 currentEpoch = parameters.getEpoch();
+                numAssets = numAssets + 1;
+
+                collections[numAssets].id = numAssets; 
+                collections[numAssets].name = pendingCollections[i].name; 
+                collections[numAssets].aggregationMethod = pendingCollections[i].aggregationMethod; 
+                collections[numAssets].epoch = currentEpoch;
+                collections[numAssets].creator = pendingCollections[i].creator;
+                collections[numAssets].repeat = pendingCollections[i].repeat;
+                collections[numAssets].assetType = uint256(assetTypes.Collection);
+                collections[numAssets].active = true;
+                uint256[] memory jobIDs = pendingCollections[i].jobIDs;
+                for(uint256 j = 0; j < jobIDs.length; j++){
+                    collections[numAssets].jobIDs.push(jobIDs[j]);
+                    collections[numAssets].jobIDExist[jobIDs[j]] = true;
+                    pendingCollections[i].jobIDExist[jobIDs[j]] = false;
+                }
+                
+                delete (pendingCollections[i]);
+
+                numPendingCollections = numPendingCollections - 1;
+
+                activeAssets.push(numAssets);
+                
+                emit CollectionActivityStatus(
+                    numAssets,
+                    currentEpoch,
+                    collections[numAssets].active,
+                    block.timestamp
+                );
+
+            }    
+        }
+    }
+
+    function deactivateAssets() 
+        external
+        onlyRole(parameters.getAssetConfirmerHash()) 
+    {
+        uint256 epoch = parameters.getEpoch();
+        
+        for(uint256 i = 0; i < pendingAssetDeactivation.length; i++){
+            uint256 id = pendingAssetDeactivation[i];
+
+            if(jobs[id].assetType == uint256(assetTypes.Job)){
+                jobs[id].active = false;
+                emit JobActivityStatus(
+                id,
+                epoch,
+                jobs[id].active,
+                block.timestamp
+                );
+            }
+            else{
+                collections[id].active = false;
+
+                for(uint j = 0; j < activeAssets.length; j++){
```

Comment:
avoid O(m*n)

Reply:
I am having to do it in this manner as if a collection is deactivated, it must be removed from activeAssets list and since we don't know the position of the id in the array, i am having to run a for loop to delete it from the list.